astropy/time/tests/test_methods.py

```diff
@@ -620,6 +630,97 @@ def test_sort(self, use_mask):
+    def test_mean(self, use_mask, axis, where):
+        self.create_data(use_mask)
+
+        kwargs = dict(axis=axis, where=where, keepdims=True)
+
+        if use_mask == 'masked':
+            if not np.isscalar(where):
+                with pytest.raises(ValueError):
+                    self.t0.mean(**kwargs)
+                with pytest.raises(ValueError):
+                    self.t1.mean(**kwargs)
+                with pytest.raises(ValueError):
+                    self.t2.mean(**kwargs)
+                return
+
+        def is_consistent(time):
+
+            kw = kwargs.copy()
+
+            where_broadcasted = np.broadcast_to(where, time.shape)
+            if use_mask == 'masked':
+                where_broadcasted = where_broadcasted & ~time.mask
+                del kw['where']
+
+            divisor = where_broadcasted.sum(axis=axis, keepdims=True)
+
+            if np.any(divisor == 0):
+                with pytest.raises(ValueError):
+                    time.mean(**kwargs)
+                return True
+
+            else:
+                time_mean = time.mean(**kwargs)
+                time_expected = Time(
+                    *day_frac(time.tai.jd1.sum(**kw), time.tai.jd2.sum(**kw), divisor=divisor),
+                    format='jd',
+                    scale='tai',
+                )
+                time_expected._set_scale(time.scale)
+                return np.all(time_mean == time_expected)
+
+        assert is_consistent(self.t0)
+        assert is_consistent(self.t1)
+
+        if axis in [None, 2]:
+            with pytest.raises(ValueError):
+                self.t2.mean(**kwargs)
+        else:
+            assert is_consistent(self.t2)
+
+    def test_mean_precision(self, use_mask):
+
+        scale = 'tai'
+        epsilon = 1 * u.ns
+
+        t0 = Time('2021-07-27T00:00:00', scale=scale)
+        t1 = Time('2022-07-27T00:00:00', scale=scale)
+        t2 = Time('2023-07-27T00:00:00', scale=scale)
+
+        t = Time([t0, t2 + epsilon])
+
+        if use_mask == 'masked':
+            t[0] = np.ma.masked
+            assert t.mean() == (t2 + epsilon)
+
+        else:
+            assert t.mean() == (t1 + epsilon / 2)
+
+    def test_mean_dtype(self, use_mask):
+        self.create_data(use_mask)
+        with pytest.raises(ValueError):
+            self.t0.mean(dtype=int)
+
+    def test_mean_out(self, use_mask):
+        self.create_data(use_mask)
+        with pytest.raises(ValueError):
+            self.t0.mean(out=Time(np.zeros_like(self.t0.jd1), format='jd'))
+
+    def test_mean_tai(self, use_mask):
+        if use_mask == 'not_masked':
+            t = Time(['2012-06-30 23:59:60.000', '2012-07-01 00:00:01.000'])
+            mean_expected = t[0] + (t[1] - t[0]) / 2
+            # mean_test = Time(t.tai.jd1.sum() / 2, t.tai.jd2.sum() / 2, scale='tai', format='jd').utc
+            mean_test = t.mean()
+            mean_expected.format = mean_test.format = 'iso'
```

Comment:
No need to set the format or test the precision below; those are all just for conversion to string; they do not influence the equality test you have.

Though if you're up to it, I would explicitly test that `t.mean() == Time('2021-07-01 00:00:00')` and add a comment that this is to check leap second behaviour. Indeed, you could even include a test that
```
# Check that leap second is dealt with correctly: for UTC, across a leap
# second bounday, one cannot just average jd, but has to go through TAI.
t.mean() != Time(*day_frac(t.jd1.sum(), t.jd2.sum(), divisor=2), format='jd')
```